include/mbedtls/fsio.h

```diff
@@ -0,0 +1,255 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(MBEDTLS_FS_IO) && !defined(MBEDTLS_FS_IO_ALT)
+/* If MBEDTLS_FS_IO is enabled then file IO functions should be made available
+ * via standard library or platform specific implementation or
+ * user defined alternative implementation. This file provides implementations
+ * based on serialization and standard library. */
+
+/**
+ * Structure returned by mbedtls_stat().
+ *
+ * \note Future versions of the library may add more fields at the end of this
+ *       structure.
+ */
+typedef struct mbedtls_stat_tag
+{
+    uint16_t    type;   /** File type */
+/** A file that can be accessed as a stream, i.e. opened with mbedtls_fopen().
+ */
+#define MBEDTLS_FSIO_DT_FILE    0
+/** A directory that can be opened with mbedtls_opendir(). */
+#define MBEDTLS_FSIO_DT_DIR     1
+/** A directory entry that is recognized neither as a stream file nor as a
+ * directory. */
+#define MBEDTLS_FSIO_DT_OTHER   2
+} mbedtls_stat_t;
+
+/* Default and alternative implementation specific interfaces. */
+
+/* MBEDTLS_SERIALIZE_C replaces Standard library FS implementation */
+#if defined(MBEDTLS_SERIALIZE_C)
+
+/**
+ * file handle.
+ *
+ */
+typedef int32_t mbedtls_file_t;
+#define MBEDTLS_FILE_INVALID    -1
+
+/**
+ * dir handle.
+ *
+ */
+typedef int32_t mbedtls_dir_t;
+#define MBEDTLS_DIR_INVALID     -1
+
+/**
+ * Definition of reference for seek offset used by mbedtls_fseek().
+ */
+#define MBEDTLS_SEEK_SET    0
+#define MBEDTLS_SEEK_CUR    1
+#define MBEDTLS_SEEK_END    2
+
+/**
+ * \brief          Read file.
+ *
+ * \param ptr      Pointer to output buffer
+ * \param size     Size of output buffer.
+ * \param stream   File handle (mbedtls_file_t).
+ *
+ * \return         Number of bytes read.
+ */
+size_t mbedtls_fread( void *ptr, size_t size, mbedtls_file_t stream );
+
+/**
+ * \brief          Write file.
+ *
+ * \param ptr      Pointer to input buffer
+ * \param size     Bytes to write.
+ * \param stream   File handle of type mbedtls_file_t.
+ *
+ * \return         Number of bytes written.
+ */
+size_t mbedtls_fwrite( const void *ptr, size_t size, mbedtls_file_t stream );
+
+/**
+ * \brief          Reads a line from file. Follows standard C fgets interface.
+ *
+ * \param s        Pointer to output buffer.
+ * \param size     Size of buffer.
+ * \param stream   File handle of type mbedtls_file_t.
+ *
+ * \return         returns s on success, and NULL on error or
+ *                 when end of file occurs while no characters have been read.
+ */
+char * mbedtls_fgets( char *s, int size, mbedtls_file_t stream );
+
+/**
+ * \brief          Sets file position. Follows standard C fseek interface.
+ *
+ * \param stream   File handle of type mbedtls_file_t.
+ * \param offset   Offset from origin.
+ * \param origin   Position from where offset is applied.
+ *                 Value is one of MBEDTLS_SEEK_SET, MBEDTLS_SEEK_CUR, or MBEDTLS_SEEK_END.
+ *
+ * \return         returns 0 on success, and -1 on error
```

Comment:
Please split into separate `\return` sections and use `\c 0` and `\c -1`.

Reply:
Similar remarks apply to the other functions in this module. While I would prefer all these style and doxygen issues to be fixed, I don't consider them blocking, though.